tests/retrieve/test_search.py

```diff
@@ -247,12 +247,13 @@ def test_many_term_search():
 
 
 def test_optional_term_search():
-    """Test search using dict inputs. This should create an OR search between the key, value pairs in the dictionaries"""
+    """Test search using dicst inputs. This should create an OR search between the key, value pairs in the dictionaries"""
     # Note: had to be careful to not create duplicates as this currently raises an error.
     res = search(
         site="bsd",
         inlet_option={"inlet": "42m", "height": "42m"},
         name_option={"station_long_name": "bilsdale, uk", "long_name": "bilsdale"},
+        data_level={"icos_data_level": 2, "data_level": "not_set"},
```

Comment:
Just to check - is this included here to check this can be run with a dictionary input at all? Which is why we're not expecting this to change the assertions?

Reply:
I checked and the data we find has "not_set" for data level, so this search should work. None of the data we find has "icos_data_level". So some extra searches are done, but they don't find anything.

But yes, the main point was to check if passing a dictionary for `data_level` works at all.